src/controllers/trainee/Controller.ts

```diff
@@ -11,79 +14,169 @@ class TraineeController {
+                      $options: 'i'
+                  }
+              }
           ]
-        });
-      } catch (err) {
-        console.log('Inside err', err);
+      };
+
+      const options: any = {
+          skip: skip,
+          limit: limit
+      };
+
+      let sortQuery: any;
+
+      if (sort) {
+          sortQuery = {
+              name: 1,
+              email: 1
+          };
+      }
+      else {
+          sortQuery = {
+              createdAt: -1
+          };
       }
-    }
 
-   delete(req: Request, res: Response, next: NextFunction) {
+      const user = new UserRepository();
       try {
-        console.log('Inside delete method of Trainee Controller');
-        res.send({
-          message: 'Trainees deleted successfully',
-          data: [
-            {
-              name: 'Trainee1',
-              address: 'Noida'
-            }
-          ]
+        const data = await user.getallTrainee(query, options, sortQuery);
+
+        res.status(200).send({
+            status: 'ok',
+            message: 'Fetched successfully',
+            Trainees: { data }
         });
-      } catch (err) {
-        console.log('Inside err', err);
       }
-    }
+          catch (err) {
+              next({
+                  message: 'Trainee Not Found',
+                  status: 404,
+                  error: err
+              });
+          }
+      }
+
+    public async create(req: IRequest, res: Response, next: NextFunction) {
+      const {  name, email, role, password } = req.body;
+      const user = new UserRepository();
+      const creator = req.userData._id;
+              try {
+                await user.create({  name, email, role, password }, creator);
+
+                res.status(200).send({
+                    message: 'User Created Successfully!',
+                    data: {
+                        'name': name,
+                        'email': email,
+                        'role': role,
+                        'password': password
+
+                    },
+                    code: 200
+                });
+              }
+              catch (err) {
+                  next({
+                      message: 'User not created',
+                      code: 404,
+                      error: err
+                  });
+              }
+      }
+      public update(req: IRequest, res: Response, next: NextFunction) {
+        const { id, dataToUpdate } = req.body;
+        const user = new UserRepository();
+        const updator = req.userData._id;
+        user.updateUser(id, dataToUpdate, updator)
+            .then((result) => {
+
+                res.status(200).send({
+                    message: 'User Updated',
+                    code: 200
+                });
+            })
+            .catch((err) => {
+                next({
+                    message: 'User Not Found for update',
+                    code: 404,
+                    error: err
+                });
+            });
```

Comment:
Use async-await to handle async operations like querying DB. Also, handle if the user does not exist in DB.